src/G11n/Currency.php

```diff
@@ -0,0 +1,29 @@
+<?php
+
+declare(strict_types=1);
+
+namespace App\G11n;
+
+use App\Exceptions\InvalidCurrencyException;
+
+final class Currency
+{
+    private const ALLOWED_VALUES = [
```

Comment:
Should that be a configuration injected through the constructor as well?
I'm pretty sure we do not want that static, what if we do ScotlandPHP (GBP) and DarkmiraTour Brasil (BRL)? In fact both of our test cases are not valid here...

Reply:
@tdutrion to prevent the value from being static. he can check the value with that: https://symfony.com/doc/current/components/intl.html#currencies ?

Reply:
This will only tell us whether the currency exist, but not whether it is allowed in our current context (understand instance of the application).
For instance, DMT Brasil may want to use both BRL and USD, while ScotPHP may want to use just GBP, or maybe GBP and EUR.

Reply:
ok i see

Reply:
What would be the best method? a servix that is injected into the ValueObject constructor that has a function that lists the countries with their devices? Having as reference a multidimensional array?

Reply:
Why multidimensional array? It's a list plain and simple (EUR, USD, ...), so either an array or a service work.

Reply:
I thought it was necessary to control the currency for each country from which my idea of multidimensional array. So a simple array will suffice. On the other hand you did not tell me if my service idea injected into the ValueObject controller was an acceptable idea. Do we agree that it will just be a service with only my Currency array?

Reply:
You have no idea of the country at this point (you're only dealing with currencies, that's what a VO is about).
A simple array would therefore be enough, you can either create an object to transfer them or just use an array directly, both would work.